dataset/src/test/scala/frameless/CollectTests.scala

```diff
@@ -59,5 +59,5 @@ class CollectTests extends TypedDatasetSuite {
 
 object CollectTests {
   def prop[A: TypedEncoder : ClassTag](data: Vector[A])(implicit c: SQLContext): Prop =
-    TypedDataset.create(data).collect().run().toVector ?= data
+    TypedDataset.create(data).collect[Job]().run().toVector ?= data
```

Comment:
Is it possible have it default to Job without specifying the type parameter?
```TypedDataset.create(data).collect[Job]().run().toVector```

that is, change `collect[Job]()` back to `collect()` and default to using `Job` in this case?

Reply:
It would be possible by having `F`-specialized instances for syntax. It isn't clear what trade off we prefer – ask users to do additional imports or specify extra type parameters.

Reply:
@imarios @kanterov I'll certainly document this, but for the benefit of discussion:
- I added the `SparkDelay[Job]` instance on the `frameless.syntax` package object. This way, the syntax import (which I assume most users already have) brings in the default instance
- I also added the default instance to the `TypedDatasetSuite` so we wouldn't have to import `frameless.syntax._` in all tests and to minimize diff noise

In this case, I had to specify the type explicitly because  the `prop` function is in the `CollectTests` companion object and doesn't have the implicit in scope. I could add the import if you think that is clearer.

Reply:
Hi @iravid, I think it's cleaner if we add the import. thank you!